lib/sqlsync-solid-js/src/hooks.ts

```diff
@@ -0,0 +1,137 @@
+import {
+  ConnectionStatus,
+  DocId,
+  DocType,
+  ParameterizedQuery,
+  QuerySubscription,
+  Row,
+  SQLSync,
+  normalizeQuery,
+  pendingPromise,
+} from "@orbitinghail/sqlsync-worker";
+import { Accessor, createEffect, createSignal, onCleanup, useContext } from "solid-js";
+import { SQLSyncContext } from "./context";
+
+export function useSQLSync(): Accessor<SQLSync> {
+  const [sqlSync] = useContext(SQLSyncContext);
+  if (import.meta.env.DEV && !sqlSync()) {
+    throw new Error(
+      "could not find sqlsync context value; please ensure the component is wrapped in a <SqlSyncProvider>"
+    );
+  }
+
+  return () => {
+    const sqlSyncValue = sqlSync();
+    if (import.meta.env.DEV && !sqlSyncValue) {
+      throw new Error(
+        "could not find sqlsync context value; please ensure the component is wrapped in a <SqlSyncProvider>"
+      );
+    } else if (!sqlSyncValue) {
+      console.error(
+        "could not find sqlsync context value; please ensure the component is wrapped in a <SqlSyncProvider>"
+      );
+    }
```

Comment:
You can remove the first if clause here and just print the error whenever it's not defined.

Reply:
done